Source/Core/DolphinWX/GameListCtrl.cpp

```diff
@@ -632,115 +668,171 @@ void CGameListCtrl::SetColors()
+
+  std::list<std::string> removed_paths;
+  std::set_difference(cached_paths.cbegin(), cached_paths.cend(), search_results.cbegin(),
+                      search_results.cend(), std::back_inserter(removed_paths));
+
+  std::vector<std::string> new_paths;
+  std::set_difference(search_results.cbegin(), search_results.cend(), cached_paths.cbegin(),
+                      cached_paths.cend(), std::back_inserter(new_paths));
+
+  const Core::TitleDatabase title_database;
+  // For now, only scan new_paths. This could cause false negatives (file actively being written),
+  // but otherwise should be fine.
+  for (const auto& path : removed_paths)
+  {
+    auto it = std::find_if(
+        m_cached_files.cbegin(), m_cached_files.cend(),
+        [&path](const std::shared_ptr<GameListItem>& file) { return file->GetFileName() == path; });
+    if (it != m_cached_files.cend())
+      m_cached_files.erase(it);
+  }
+  for (const auto& path : new_paths)
+  {
+    auto file = std::make_shared<GameListItem>(path, title_database);
+    if (file->IsValid())
+      m_cached_files.push_back(std::move(file));
+  }
+
+  for (auto& file : m_cached_files)
+    file->ReloadINI();
+
+  // Post UI event to update the displayed list
+  QueueEvent(new wxCommandEvent(DOLPHIN_EVT_REFRESH_GAMELIST));
+
+  post_status(_("Scan complete!"));
+
+  SyncCacheFile(true);
 }
 
-void CGameListCtrl::OnLocalIniModified(wxCommandEvent& ev)
+void GameListCtrl::OnRefreshGameList(wxCommandEvent& WXUNUSED(event))
 {
-  ev.Skip();
-  std::string game_id = WxStrToStr(ev.GetString());
-  // NOTE: The same game may occur multiple times if there are multiple
-  //   physical copies in the search paths.
-  for (std::size_t i = 0; i < m_ISOFiles.size(); ++i)
-  {
-    if (m_ISOFiles[i]->GetGameID() != game_id)
-      continue;
-    m_ISOFiles[i]->ReloadINI();
+  RefreshList();
+}
 
-    // The indexes in m_ISOFiles and the list do not line up.
-    // We need to find the corresponding item in the list (if it exists)
-    long item_id = 0;
-    for (; item_id < GetItemCount(); ++item_id)
-    {
-      if (i == static_cast<std::size_t>(GetItemData(item_id)))
-        break;
-    }
-    // If the item is not currently being displayed then we're done.
-    if (item_id == GetItemCount())
-      continue;
+void GameListCtrl::OnRescanGameList(wxCommandEvent& WXUNUSED(event))
+{
+  m_scan_trigger.Set();
+}
 
-    // Update all the columns
-    for (int j = 1; j < NUMBER_OF_COLUMN; ++j)
-    {
-      // NOTE: Banner is not modified by the INI and updating it will
-      //  duplicate it in memory which is not wanted.
-      if (j != COLUMN_BANNER && GetColumnWidth(j) != 0)
-        UpdateItemAtColumn(item_id, j);
-    }
-  }
+void GameListCtrl::OnLocalIniModified(wxCommandEvent& ev)
+{
+  ev.Skip();
+  // We need show any changes to the ini which could impact our columns. Currently only the
+  // EmuState/Issues settings can do that. We also need to persist the changes to the cache - so
```

Comment:
"EmuState/Issues" is misleading IMO. You should either mention EmuState as a whole or mention all of the three things that can be in EmuState, not something in-between. Otherwise this comment gives the impression that the EmuState Title can't affect the columns.